pyamo/review.py

```diff
@@ -12,14 +12,35 @@
 import traceback
 
 from zipfile import ZipFile
-from urlparse import urlparse, urljoin
+from six.moves.urllib.parse import urlparse, urljoin
 from mozprofile import FirefoxProfile
 
 from .utils import AMO_BASE, AMO_EDITOR_BASE, AMO_REVIEWERS_API_BASE, csspath
 from .lzma import SevenZFile
 
 import lxml.html
-import magic
+try:
+    import magic
+    archiveCtorMapping={
+        "application/x-7z-compressed": SevenZFile,
+        "application/zip": ZipFile
+    }
+    def getArchiveCtor(path):
+        return archiveCtorMapping[magic.from_file(path, mime=True)]
+except:
+    archiveCtorMapping={
+        "application/x-7z-compressed": SevenZFile,
+        "application/zip": ZipFile
+    }
+    def getArchiveCtor(path):
```

Comment:
I was thinking about something like this too. I'm fine with removing the magic code alltogether and only using the manual method. Make sure to adapt setup.py as well.